### DISCUSSION TITLE
当使用同步助手后反复更改配置无故不生效的解决方案

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3696] [重要通知]
## 配置项

这是下面的内容会用到的一个词，包括：阅读模式右下角触发器的选项、选项页的选项、插件的选项等一切可以改动的内容。

## 问题现象

当使用同步助手在反复修改选项页或者阅读模式的配置刷新后不生效，一些常见场景：

1. 在选项页反复修改某个配置项，尤其是定制 Markdown
2. 在不同的阅读模式页面下，反复修改某个配置项产生

## 问题原因

1. 每个阅读模式都保存同一份配置文件，当修改后自动保存，就会出现反复覆盖配置项的情况，类似「同一个文本被不同的编辑器同一时间反复修改」的情况。

2. 为了提高同步助手的配置文件保存性能，不是每次操作都会保存，每次保存有个时间差的间隔（很短， 0.2m 左右且有算法支持），但还是存在一定误保存率，正常使用完全没问题，仅出现在频率保存配置项时。

## 解决方案

1. 不要同时多开多个 Tab 同时操作配置项，因为会互相覆盖配置。

2. 只开两个：一个选项页（A）；一个进入阅读模式的页面（B），即可。当 A 改动后刷新 B使其使用最新的数据；反之也一样。

3. 使用同步助手后，如果非常频繁的保存和设置项，就先关闭自动同步功能，等配置稳定了（毕竟不需要随时改动，只是配置一次），在开启自动同步，然后覆盖本地配置文件。（下图）

   ![image](https://user-images.githubusercontent.com/81074/162602933-848b29d3-fcb0-49a6-bfe4-447f2fd299b6.png)

4. 如果经常在输入框修改内容的话，尤其是多行输入框（常见场景：Markdown 模板）建议在其它文本编辑器修改，然后在粘贴回去，且要在输入框改动一次，刷新页面后确保生效。

5. 使用同步助手的用户，建议常开坚果云，当出现任何配置的问题时，利用坚果云的历史记录，可以确保还原到任意节点。

## 插件不生效（或没有插件触发器的用户）

这是一个简悦的 Bug，仅当使用同步助手的自动同步时出现，但并不会影响每个使用同步助手的用户，当出现此问题时：

1. 通过 [此教程](https://github.com/Kenshin/simpread/discussions/2342)  解决。

2. 暂时关闭自动同步，等插件设置完毕后再开启此功能。

## 备注

不使用同步助手的用户，也建议使用看下**解决方案**中的步骤 ➊ ➋ ➍

**Comment by zhcheng828** [comment]
我发现解决方案中步骤3在开启同步助手之前，要先删除一下本地的配置文件，不然还是会被本地覆盖

**Reply by Kenshin** [reply]
不需要的，因为截图上已经说明了需要覆盖本地。（当然，也可以删除它）
